### ISSUE TITLE
Manually subscribe to collective updates

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#1445]
## User story  
As an OpenCollective user, I would like to be able to subscribe to some collective's updates without having to donate to them. Sometimes you may be interested in the evolution of a collective, but not ready to finance it yet.

It could also be a way to engage users not registered yet on OpenCollective, who are not ready to donate  but who may be interested in getting updates from the collective.

## Best solution for this problem
Add a `subscribe to updates` / `unsubscribe` button (with a bell icon?) on the collective page.

## Metrics 
How much users subscribe without donating.

**Comment by xdamman** [comment]
Yes, yes, yes.
There is a hack which is creating a free tier, but this is something we should design for /cc @cuiki 
Maybe add a "subscribe" or "follow" or "join" button as the primary call to action?

**Comment by piamancini** [comment]
Please make sure you implement this as an opt-in or opt-out for collectives. I am working with collectives to have subscriptions that have as a perk  previews of releases via the updates feature.

**Comment by alannallama** [comment]
We got a user request for this feature today. I think it's a good one.

**Comment by alannallama** [comment]
That's a good question @Betree - on Patreon, private updates are only for paying backers. Have we gotten any clue from users if this kind of 'perk' is the reason they want private updates? I would guess yes, because if anyone can sign up for updates and get the private ones, there's no point in them being private.

**Comment by Betree** [comment]
@alannallama Yes, it is the reason why they want private updates. We actually got this feature request from the Dokku collective and we're co-financing the bounty for it so it's something important to them.

From what we discussed yesterday with Pia, (2) seems like the easiest and more logical solution so it's the preferred one right now.

It could also be helpful to have a tooltip above the follow button with a message like:

> You will receive all the public updates from {collective}. Become a backer to also receive the private updates!

**Comment by alannallama** [comment]
Yep, sounds good. If you're not sure of the design or wording, have a look
at how Patreon explains backer-only posts.

--
Alanna Irving
hello@alanna.space
https://alanna.space

**Comment by Relequestual** [comment]
Still want to offer this to our users!

**Comment by xdamman** [comment]
This looks like a highly requested feature. Also in https://github.com/opencollective/community/issues/2
Maybe worth considering for a bounty @znarf ?

**Comment by josegonzalez** [comment]
The Dokku collective would be willing to sponsor/co-sponsor a bounty for this issue. Feel free to reach out to me via email/slack (I'll re-add the OC slack tomorrow).

**Comment by alanna** [comment]
Thank you @josegonzalez - Do you know any developers who have the skills/capacity to work on this? Open Collective could co-sponsor the bounty funding, but the real bottleneck for us is dev capacity.

**Comment by josegonzalez** [comment]
Not sure I know anyone personally but can dig through contacts :)

Is there a tag that we can add to this to show that it has a bounty or something? Might be a good way to raise awareness.

**Comment by bolariin** [comment]
Can I work on this?

**Comment by Betree** [comment]
@bolariin sorry for the late reply, feel free to work on it! This issue is quite complex, so feel free to ask for help here on on [Slack](https://opencollective.slack.com/archives/C0RMV6F8C) if ever needed.

**Comment by Betree** [comment]
Hi @bolariin, any news on this one, can we do anything to help?

**Comment by bolariin** [comment]
@Betree Sorry no news on this. I had started but the last couple of months have been busy and I let this slip.

**Comment by Betree** [comment]
@bolariin No problem :slightly_smiling_face: I'll de-assign you in case someone wants to tackle it, but feel free to re-assign yourself if you want to go back to it later

**Comment by alanna** [comment]
I still think this is a great idea but I'm not sure how to prioritise it. Maybe worth doing some scoping?

**Comment by snitin315** [comment]
@Betree Can I work on this one?

**Comment by Betree** [comment]
@snitin315 Thanks for proposing, we'll need to discuss it internally first to make sure we have the bandwidth to review and maintain. I'll let you know.

**Comment by alanna** [comment]
I realised this is the issue blocking us from switching from Mailchimp for host Updates for OSC and OCF. So I'm adding `important` because I think it would be great to use our own systems for these important comms.

The other issue on my mind for that is deliverability and whether we'll get through spam filters as well as mailchimp, but that's another issue.

**Comment by Betree** [comment]
We internally discussed an important blocker for this feature: while we want to keep this free for posting updates to your contributors, **there will be a real infrastructure cost for us if people start using this as a public newsletter service** and we don't set some kind of limitations or pricing first, as it will basically become a free version of Mailchimp. An update sent to 10.000 persons is at least a $8 USD charge for us. We need to think about ways to prevent abuse in order to make this change possible. Hopefully, something with a minimal impact on existing collectives. 

The simplest I can think of right now, without locking us to a specific model is to have a tolerance based on their number of contributors (eg. if you have 10,000 contributors, it's ok to send 12,000 emails but it's not ok to send 12,000 emails if you only have 30 contributors). We can then ask collectives to contact us if they reach that limit to find an agreement, preventing them from publishing more updates before the situation is resolved.

Technically, we could achieve an MVP with:
- Adding a new `emailsSentCount` column on `Updates`, increment this number when the update gets published
- Before publishing a new update, check the total number of updates already published for this month. If the total number is more than 150% (we need to think of a number that makes sense) of the total number of contributors to the collective,(`SUM(emailsSentCount)` + nbEmailsToSend > `nbUpdates` * `nbContributors` * 1.5), prevent publishing the update and ask collective admins to contact support
- When/if we receive such a request, we can directly see with the collective for them to setup a contribution to Open Collective and set a special flag on their collective to remove the limitation
- We should update the docs/FAQ to explain all that

As a longer-term solution, we can imagine surfacing the pricing before publishing the update and automatically charge the collective when it gets published (again, only when the number of emails is above the tolerance range, based on the number of contributors).

**Comment by alanna** [comment]
It seems like a good enough approach for now. I really want us to find a good MVP solution to get the feature out there in the wild and see how people actually use it. But it's much easier to make a restricted feature more open than go the other way (i.e. releasing it to everyone and then later saying they have to pay), so I think starting with this limitation in place is a good idea. I also like how charging for actual cost of use is straightforward and doesn't need to complicate the pricing model. 

We could automate it so the system submits an expense to their budget. I think we should have a lower threshold too - I don't want to bother charging people <$1 amounts every time they send a newsletter. So maybe we only charge when it costs us over a certain amount? 

We should keep in mind the benefits their usage has in addition to the cost. If someone develops a large mailing list and that exposes thousands of people to the platform, it would only take a few of those people becoming contributors or starting Collectives for it to easily pay for itself. It's just an unknown. But regardless, we need to prevent abuse, such as people using this for spamming, or just trying to get a free mailchimp without ever intending to do fundraising.

**Comment by jasikpark** [comment]
Perhaps it's possible to generate an RSS feed for the updates, and recommend piping that into a newsletter service that they setup themselves? I just want some RSS xD

**Comment by Betree** [comment]
@jasikpark I'd love to have RSS too, that feature request has actually been documented here: https://github.com/opencollective/opencollective/issues/3504. But not everyone uses RSS, we need something simple that works for most users.

**Comment by 0xdevalias** [comment]
As a user who's wanted this sort of a feature in the past (and has so far worked around it by making small $1-5 contributions to projects I want to 'follow'), I would find value in this feature even if it didn't lead to me getting emailed out updates.

Just the ability to log into my OpenCollective account and go to a page of 'projects i'm following' would be a useful start, as there are a lot of cool projects out there that I come across and then forget exist over time. Of course I could work around that by just bookmarking them in my browser.. but that generally just ends up being like throwing the link into the void never to be resurfaced. I also tend to find that I (and others I know) can often use the pages/projects we follow as part of showing what we care about /communicating that to others when they are able to be seen from our profile; in the same way that our current contributions can be seen/displayed from our profiles.

The 'next step up' from there for me would be having a sort of in-platform notification system, so that I could see any updates from projects that i've followed and whether I had seen/read them yet or not.

Then finally, email updates would sort of be a 'last tier' for me, definitely can see the value in it, but this feature would still be really useful to me without having it at all. One potential thought I had here was that instead of taking on the financial costs/risk of sending the emails, perhaps the feature could be designed in a way that various 'mail as a service' providers could be 'plugged in' as options the collective could use, and then link to their own accounts with that provider. Initially perhaps only 1 is supported, but if designed well, then it would be a much smaller piece of work for others to contribute different 'connectors' to other providers in future, if/as needed. This same system could potentially be designed in a way that it could work for sending the updates via other methods as well (eg. Slack, Telegram, etc). Then, to bring it back to the previous discussions around OpenCollective providing the email service 'built in', this could just end up being a special 'limited' version of one of these connectors based on whatever volume/cost calculations seem feasible/sustainable.

So in summary, I wonder if there are some ways to break this down into smaller chunks of work that can still provide value, even while some of the bigger blocking questions might remain unknown?

Just my /2c, perspectives, and musings. Hopefully they're helpful :)

**Comment by Betree** [comment]
@0xdevalias thanks for that feedback, I love the idea of having a place to track all the initiatives you follow. In-app notification management would definitely be a huge plus for this feature, it's something we're discussing in https://github.com/opencollective/opencollective/issues/5382.

**Comment by BenJam** [comment]
>  I love the idea of having a place to track all the initiatives you follow

this is a consideration for #5501 too.

**Comment by simonv3** [comment]
This would really help our mutual aid group out. We've got key volunteers who don't (because they can't) donate financially but they're still contributors, but because they don't donate they don't get emails from us. If we could add them manually or have a sign-up sheet that would be great. Having this in one spot would save us a lot of headache from having to set up a separate mailing list or something like that.

**Comment by Memo-Es** [comment]
I really like the challenge presented in this issue! I think it needs some design exploration to flesh out a coherent MVP and shape the other little things that complete the experience: Managing subscriptions, the concept of membership levels for communities (And stop calling everyone just contributors, among other things) Big +1!

**Comment by jasikpark** [comment]
"Follower" vs "Patron" is useful and helpful in patreon, though that's maybe a different paradigm
